### PR TITLE
Add setup and upgrade docs

### PR DESCRIPTION
**Pull request by beerose**

### CODE REVIEW

File: app/pages/docs/blitz-with-next.mdx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+```sh
+yarn add blitz@alpha @blitzjs/next
+
+# or
+
+pnpm add blitz@alpha @blitzjs/next
+
+#or
+
+npm i blitz@alpha @blitzjs/next
```
Comment (dillonraphael) [inline app/pages/docs/blitz-with-next.mdx:21]: @blitzjs/next@alpha

File: app/pages/docs/upgrading-from-framework.mdx [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+```js
+const { withBlitz } = require("@blitzjs/next")
+
+module.exports = withBlitz()
+```
+
+### Update dependencies in `package.json` {#update-dependencies}
+
+1. Update `react`, `react-dom` to the latest versions.
+2. Install `@blitzjs/next`, `@blitzjs/rpc`, and `@blitzjs/auth`.
```
Comment (dillonraphael) [inline app/pages/docs/upgrading-from-framework.mdx:53]: `@blitzjs/next@alpha`, `@blitzjs/rpc@latest`, `@blitzjs/auth@latest`

File: app/pages/docs/upgrading-from-framework.mdx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+Now, for most of the things previously imported from `blitz` package,
+you'd need to update the import to the new packages. Use the following
+list as a reference:
+
+| Import                        | Source Package  |
+| ----------------------------- | --------------- |
+| `GetServerSideProps`          | `next`          |
+| `InferGetServerSidePropsType` | `next`          |
+| `GetServerSidePropsContext`   | `next`          |
+| `useRouterQuery`:             | n`xt/router`    |
```
Comment (dillonraphael) [inline app/pages/docs/upgrading-from-framework.mdx:68]: oops

File: app/pages/docs/upgrading-from-framework.mdx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+    }),
+    BlitzRpcPlugin(),
+  ],
+})
+```
+
+### Create API Route for the Zero-API Layer {#create-api-route}
+
+To setup the Zero-API layer, you'd need to create a
+`pages/api/rpc/[[blitz]].ts` file with the followinf content:
```
Comment (dillonraphael) [inline app/pages/docs/upgrading-from-framework.mdx:178]: following

File: app/pages/docs/blitz-with-next.mdx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+## Blitz server setup {#blitz-server-setup}
+
+If you want to use Blitz's server functionalities like auth, middlewares,
+rpc, you'd need to create a `blitz-server.ts` file somewhere in your
```
[the remark this answers is not in the document]
Comment (beerose) [reply app/pages/docs/blitz-with-next.mdx:27]: it doesn't have to be app/

**Review by dillonraphael — APPROVED** [review]

**Pull request merged**